lib/luaT/luaT.c

```diff
@@ -497,27 +496,65 @@ int luaT_classmodulename(const char *tname, char *module_name)
   return tname[n] == '.';
 }
 
-/* Lua only functions */
+void traverse_tables(lua_State *L, const char *tname, int *L_idx)
```

Comment:
In fact, you can just use relative indices in this function, couldn't you? Then you don't need to pass in the index at all. Or the function get it itself via `lua_gettop`.